src/app.jsx

```diff
@@ -198,53 +215,85 @@ export default class App extends React.Component {
     });
 
     return (
-      <TabContainer activeKey={activeTab}>
-        <Navbar onDragOver={dragOverHandlerNone}>
-          <Navbar.Brand onDragOver={dragOverHandlerNone}>
-            <Nav.Link
+      <React.Fragment>
+        <DataDownloadModal
+          show={showDownloadModal}
+          closeModal={() => this.showDownloadModal(false)}
+          storeDownloadDir={this.storeDownloadDir}
+        />
+        <TabContainer activeKey={activeTab}>
+          <Navbar onDragOver={dragOverHandlerNone}>
+            <Navbar.Brand onDragOver={dragOverHandlerNone}>
+              <Nav.Link
+                onSelect={this.switchTabs}
+                eventKey="home"
+                onDragOver={dragOverHandlerNone}
+              >
+                InVEST
+              </Nav.Link>
+            </Navbar.Brand>
+            <Nav
+              variant="pills"
+              className="mr-auto horizontal-scroll"
+              activeKey={activeTab}
               onSelect={this.switchTabs}
-              eventKey="home"
               onDragOver={dragOverHandlerNone}
             >
-              InVEST
-            </Nav.Link>
-          </Navbar.Brand>
-          <Nav
-            variant="pills"
-            className="mr-auto horizontal-scroll"
-            activeKey={activeTab}
-            onSelect={this.switchTabs}
-            onDragOver={dragOverHandlerNone}
-          >
-            {investNavItems}
-          </Nav>
-          <LoadButton
-            openInvestModel={this.openInvestModel}
-            batchUpdateArgs={this.batchUpdateArgs}
-          />
-          <SettingsModal
-            className="mx-3"
-            saveSettings={this.saveSettings}
-            investSettings={investSettings}
-            clearJobsStorage={this.clearRecentJobs}
-          />
-        </Navbar>
-
-        <TabContent id="top-tab-content">
-          <TabPane eventKey="home" title="Home">
-            <HomeTab
-              investList={investList}
+              {investNavItems}
+            </Nav>
+            {
+              (downloadedNofN)
+                ? (
+                  <DownloadProgressBar
+                    downloadedNofN={downloadedNofN}
+                    expireAfter={5000} // milliseconds
+                  />
+                )
+                : <div />
+            }
+            <LoadButton
               openInvestModel={this.openInvestModel}
-              recentJobs={recentJobs}
+              batchUpdateArgs={this.batchUpdateArgs}
             />
-          </TabPane>
-          {investTabPanes}
-        </TabContent>
-      </TabContainer>
+            {
+              // don't render until after we fetched the data
+              (investSettings)
+                ? (
```

Comment:
Is this to avoid a user clicking the download button and bringing up the download modal while it's currently working?

Reply:
No it's not strictly related to this PR. I think I hit the same problem you did a while back, the `SettingsModal` was rendered at least once before it's props (`investSettings`) were fetched. And so `SettingsModal` was being called with unexpected data. It meant having to wait to `find` elements in tests, which is fine, but I think this is better since there is no good that can come from rendering `SettingsModal` without `investSettings`.

Reply:
Ah, right I see. When I read `don't render until after we fetched the data` I was in the mind set of download data.